ui/perfherder/compare/CompareTable.jsx

```diff
@@ -79,6 +91,58 @@ export default class CompareTable extends React.PureComponent {
     }
   };
 
+  // Sorts the specified key of the array data depending on the typeof key
+  // If something changes in columns, you just need to update sortableKeys
+  // with the corresponding type
+  // The data model of the sortable keys is:
+  // name: string
+  // originalValue: number
+  // newValue: number
+  // deltaPercentage: number
+  // confidence: number
+  // originalRuns: array
+  // newRuns: array
+  // default: string
+  sortBy = (columnName, data) => {
+    let { sortOrder, sortClass } = this.state;
+    const { sortedTestName, sortedColumnName } = this.state;
+    const { testName } = this.props;
+    const sortableKeys = {
+      name: 'string',
+      originalValue: 'number',
+      originalStddevPct: 'number',
+      newValue: 'number',
+      newStddevPct: 'number',
+      deltaPercentage: 'number',
+      confidence: 'number',
+      originalRuns: 'array',
+      newRuns: 'array',
+    };
+    const sorted = true;
+    if (sorted) {
+      // ^ sorted flag becomes true after clicking for the first time any column
+      // Toggle sort order if clicked a column header several consecutive times
+      if (sortedTestName === testName && sortedColumnName === columnName) {
+        data.sort(sortHashes(columnName, sortableKeys[columnName], !sortOrder));
```

Comment:
We want to avoid mutating state directly (here or elsewhere), so instead of doing this you should use spread syntax to create a copy of `data` and then update it in `setState`.

Reply:
```
  componentDidUpdate(prevState) {
    const { sortedData } = this.state;
    if (this.state.sortedData !== prevState.sortedData) {
      this.setState(sortedData);
    }
  }
```
`ESLine: Do not use setState in componentDidUpdate(react/no-did-update-set-state)`

Reply:
hrmm, I think it is fine in this case. I disable this rule often - its one of the few I do - because it is poorly written and fails to recognize if you're calling set state in a condition (and the condition doesn't result in render thrashing or some other undesired effect). The docs support this action: https://reactjs.org/docs/react-component.html#componentdidupdate

So, this rule is fine to disable.

Reply:
Read the documentation and I also that some rules are poorly written. But you're the experienced js dev here, thanks for answering!